### ISSUE TITLE
Got Validation Failed error while running the pipeline (Bundler + Gemfile approach)

### ISSUE THREAD

**Issue by tatagaziz** [jonabc/licensed-ci#127]
Got an error while running the pipeline using bundler + Gemfile approach. 
```
/usr/bin/git remote add licensed-ci-origin ***github.com/my-private-repository/project-name

::set-output name=licenses_branch::licensed-ci-actions-licenses
##[debug]='licensed-ci-actions-licenses'

::set-output name=user_branch::licensed-ci-actions
##[debug]='licensed-ci-actions'
Error: Validation Failed: {"message":"The listed users and repositories cannot be searched either because the resources do not exist or you do not have permission to view them.","resource":"Search","field":"q","code":"invalid"}
##[debug]Node Action run completed with exit code 1
##[debug]Finishing: Run jonabc/licensed-ci@v1.5.8
```

I did a quick googling and it seems the error is related to Github Octokit, although I'm not really sure because I don't really know how to correctly read this log and pinpoint which passage that produced this error. One thing to note is that one of the gems that needs to be installed is a proprietary gem that is stored on github packages. 

I'll be happy to include additional debugging info if necessary, thanks!

**Comment by jonabc** [comment]
:wave: @tatagaziz this sounds like you're trying to access a private repo using an auth token that does not have read permissions to that repo. the most likely causes I've seen for this are:
1. the action is using the default actions runner github auth token `${{secrets.GITHUB_TOKEN}}`, which only has permissions to the repo that the action is run from
   - in this case, please create a user PAT with repository write permissions to use with the action.  The user must have access to the repo running the action as well as any private repos that the action needs to access 👍 .  Write permissions are needed on the token so that metadata files can be pushed to the repo, and issue comments or PRs can be created.
2. the action is using a user PAT and that user does not have read access to the private repo
   - in this case, please review which repos the user account corresponding to the PAT has access to and give repo access to the user as needed
   
Do either of those cases apply here?  Happy to help further if needed, and once we get this figured out I'll update the README with any additional helpful information

**Comment by tatagaziz** [comment]
Thank you for your response! It does currently use the default actions runner github auth token, I will try to issue a PAT and retry the workflow by using the PAT. I will update the situation once it is done 👍

**Comment by tatagaziz** [comment]
Hello, the PAT works and the workflow isn't catching the Validation Failed error anymore, thank you for your help! :bow:

On the other hand, it seems one of the gems installed during `run: bundle install` step is of a different version than the one in Gemfile.lock and that causes the `licensed cache` step to fail though. 
( It's a different topic than the initial issue I asked though so feel free to close this one if I have to create another issue with a different question)

**Comment by tatagaziz** [comment]
Pardon me, it seems the `bundle install` problem is a mistake on my part! (there was a settings mismatch), problem is solved and thank you once again for your help!

**Comment by jonabc** [comment]
updated in https://github.com/jonabc/licensed-ci/commit/11816b031fcc534bf433d187563c914605d2b244.  If you have any suggested improvements please feel free to open a PR with any changes! 🙇
